api/app/hfi/hfi_calc.py

```diff
@@ -64,159 +63,123 @@ def generate_station_daily(raw_daily: dict,
+        temperature=raw_daily.get("temperature", None),
+        relative_humidity=raw_daily.get("relativeHumidity", None),
+        wind_speed=raw_daily.get("windSpeed", None),
+        wind_direction=raw_daily.get("windDirection", None),
+        precipitation=raw_daily.get("precipitation", None),
+        grass_cure_percentage=raw_daily.get("grasslandCuring", None),
         ffmc=ffmc,
-        dmc=raw_daily.get('duffMoistureCode', None),
-        dc=raw_daily.get('droughtCode', None),
-        fwi=raw_daily.get('fireWeatherIndex', None),
+        dmc=raw_daily.get("duffMoistureCode", None),
+        dc=raw_daily.get("droughtCode", None),
+        fwi=raw_daily.get("fireWeatherIndex", None),
         # Danger class possible values: 1, 2, 3, 4, and 5. Where 1 is the lowest, and 5 is the highest
         # This is the same for the dangerGrassland and dangerScrub,
         # but those values aren’t really used anywhere,
         # just calculated and stored along with the forest danger rating
         # You can see current stations/rating on this page here:
         # https://wfapps.nrs.gov.bc.ca/pub/wfwx-danger-summary-war/dangerSummary
-        danger_class=raw_daily.get('dangerForest', None),
+        danger_class=raw_daily.get("dangerForest", None),
         isi=isi,
         bui=bui,
         rate_of_spread=fire_behaviour_prediction.ros,
         hfi=fire_behaviour_prediction.hfi,
-        observation_valid=raw_daily.get('observationValidInd', None),
-        observation_valid_comment=raw_daily.get('observationValidComment', None),
+        observation_valid=raw_daily.get("observationValidInd", None),
+        observation_valid_comment=raw_daily.get("observationValidComment", None),
         intensity_group=fire_behaviour_prediction.intensity_group,
         sixty_minute_fire_size=fire_behaviour_prediction.sixty_minute_fire_size,
         fire_type=fire_behaviour_prediction.fire_type,
-        error=raw_daily.get('observationValidInd', None),
-        error_message=raw_daily.get('observationValidComment', None),
-        last_updated=datetime.fromtimestamp(raw_daily['lastEntityUpdateTimestamp'] / 1000, tz=timezone.utc)
+        error=raw_daily.get("observationValidInd", None),
+        error_message=raw_daily.get("observationValidComment", None),
+        last_updated=datetime.fromtimestamp(raw_daily["lastEntityUpdateTimestamp"] / 1000, tz=timezone.utc),
     )
 
 
 def get_prep_day_dailies(dailies_date: date, area_dailies: List[StationDaily]) -> List[StationDaily]:
-    """ Return all the dailies (that's noon, or 20 hours UTC) for a given date """
+    """Return all the dailies (that's noon, or 20 hours UTC) for a given date"""
     dailies_date_time = get_hour_20_from_date(dailies_date)
     return list(filter(lambda daily: (daily.date == dailies_date_time), area_dailies))
 
 
+def get_hydrated_stations(stations: List[PlanningWeatherStation], stations_by_code: dict[int, WeatherStation]):
```

Comment:
Are the typings here correct? It looks lile `get_hydrated_stations()` includes a dictionary of `WeatherStation` objects and returns a list of `WeatherStation` objects that have different properties?

Reply:
I think there's existing types with the same names imported from API schemas that overlap with database types. I'll see if I can distinguish and correct the imports.

Reply:
There's some duplication of model definitions it looks like. I added a type alias to distinguish types in: https://github.com/bcgov/wps/pull/4421/commits/8a05976ac7fde7c3af0849f58203db7b685d5f0d

I added a dev task for a more permanent fix here: https://github.com/bcgov/wps/issues/4423